SpywareFilter/sections/specific.txt

```diff
@@ -19,6 +19,12 @@
 ||tags.tiqcdn.com/utag/*/prod/utag.js$domain=radissonhotels.com|post.ch|usnews.com|yemeksepeti.com|krymr.com
 !
 !
+! TODO: remove a modifier `~domain` when the issues below will be fixed
+! https://github.com/AdguardTeam/Scriptlets/issues/163
+! https://github.com/AdguardTeam/Scriptlets/issues/164
+! https://github.com/AdguardTeam/Scriptlets/issues/165
+||mc.yandex.ru/metrika/tag.js$script,important,redirect=metrika-yandex-tag,domain=~coddyschool.com
+||mc.yandex.ru/metrika/watch.js$script,important,redirect=metrika-yandex-watch,domain=~tv.yandex.*|~auth.wi-fi.ru
```

Comment:
Do we really need an `$important` modifier in these rules?

Reply:
Actually, yes, because we have exceptions. I think we can use hints, but, in general, it will be difficult. Maybe for some cases, but not in common.
In this case, this request will be excluded for `auth.wi-fi.ru` and the redirect rule won't work without `$important`.
```
||mc.yandex.ru/metrika/watch.js$script,redirect=metrika-yandex-watch
@@||mc.yandex.ru/metrika/watch.js$domain=auth.wi-fi.ru
```

Reply:
But `auth.wi-fi.ru` is excluded from this rule too, so redirect rule will be not applied anyway (I do not mean your example, but the real case here - https://github.com/AdguardTeam/AdguardFilters/pull/101887/files#diff-4229244108ff6695dbabadfef8e0d30d66184c10c4cb116e6d73af18009581a1R27).

Now, if `redirect=metrika-yandex-tag/metrika-yandex-watch` is causing any issue and website is broken when one of these scripts are blocked, you have to add domain in two places, one in a `redirect` rule, second in exception, or you can just add exception with `important` modifier (if I'm not wrong, it's supported only by AdGuard).
We have also redirect rules for `google-analytics`, `google-analytics-ga`, `googletagmanager-gtm`, `googletagservices-gpt` - all these rules are without `important` modifier and as far as I can see, we are adding hints (`!+ PLATFORM(ios, ext_android_cb, ext_safari)`) for these websites, if it's causing problem on platforms which do not support `redirect` modifier.

If you prefer this way, then fine.

Reply:
To be honest, I think that in general it's easier to use such rules without hints.
If you want to exclude a request only for platforms that don't support `redirect` modifier, you just add a domain to the common exception rule and the redirect rule will work as usual on supported platforms. In this case, you don't have to limit the exception rule with adding hint.

Reply:
You do not need to use hints every time, as you have added a section for exceptions, so for example, you can just add (and remove `important` from redirect rules):
```
! For all platforms
! metrika-yandex - common exceptions for tag.js, watch.js
@@||mc.yandex.ru/metrika/tag.js$domain=example.org
@@||mc.yandex.ru/metrika/watch.js$domain=example.net
!
! For platforms which do not support $redirect rules
!#if (adguard_ext_safari || adguard_app_ios || adguard_ext_android_cb)
@@||mc.yandex.ru/metrika/tag.js$domain=domain1|domain2
@@||mc.yandex.ru/metrika/watch.js$domain=domain3
!#endif
!
```
And you will not need to add `~example.org` and `~example.net` to redirect rules.
But as I said, if you prefer this way, then fine :)

Reply:
I was thinking about adding such a structure, but in general I'd prefer use it without hints.
By the way, that's one of the reasons I decided to make changes via the pull request. We can vote about it together, if you want :)

Reply:
In general I prefer to not use `important` modifier if it's not really needed.
Now you will have exceptions rules available/visible in filters which will be overridden by `redirect` rule with `important`, and I think it may be a bit misleading, also you will need to add `important` modifier to exception if you will want to test an exception rule in "User rules", but it's just my opinion.

Reply:
Ok, will change it.

Reply:
Done.